pkg/resources/external_function.go

```diff
@@ -239,39 +232,55 @@ func CreateContextExternalFunction(ctx context.Context, d *schema.ResourceData,
 			args = append(args, sdk.ExternalFunctionArgumentRequest{ArgName: argName, ArgDataType: argDataType})
 		}
 	}
+	argTypes := make([]sdk.DataType, 0, len(args))
+	for _, item := range args {
+		argTypes = append(argTypes, item.ArgDataType)
+	}
+	id := sdk.NewSchemaObjectIdentifierWithArguments(database, schemaName, name, argTypes...)
```

Comment:
nit: the loop above could be incorporated in the constructor (taking some common ArgRequest - name and type - but this is a possible improvement for the future)

Reply:
Got it, created SNOW-1629561 for that in the identifiers rework follow up